### PR TITLE
add overview sections to docs and update index

### PR DESCRIPTION
**Pull request by snopoke**
Most of the content is from here: https://docs.google.com/document/d/1bA2hOE81U7ZOy-FaYHVqQ-4ipnK4F41gdL_q3DTLoXM/edit#

### CODE REVIEW

**Review by esoergel — COMMENTED** [review]
This is a very comprehensive overview.  I left a bunch of feedback, feel free to pick and choose through it. 

These documents largely avoid using the common names of components (form processor, app builder, pillowtop, UCR, web user, mobile worker), which makes it hard to approach for audiences used to using CommCare or reading about it on the wiki.  I'm guessing it was drafted initially for review by an external technical audience, but IMO it could use some amendments if it's to serve as an introduction to CommCareHQ for developers who will be working on the project.

File: docs/overview/platform.rst [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+instance.
+
+APIs
+~~~~
+
+CommCare has extensive APIs to get data in and out for bidirectional integration with other systems. This method of
+data integration requires familiarity with RESTful HTTP conventions, such as GET and POST and url parameters.
+Character encoding is UTF-8. Dates should be compliant ISO 8601. In some cases CommCare uses RFC 3339, but is
+moving towards ISO 8601. Guides should be generated in accordance with RFC 4122. Documented APIs and use of
+Standards are readily available.
```
[the remark this answers is not in the document]
Comment (snopoke) [reply docs/overview/platform.rst]: 2fb8b398a91

File: docs/overview/platform.rst [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+MOTECH Repeaters
+~~~~~~~~~~~~~~~~
+
+For interoperability with external systems which process transactional data, CommCare has a MOTECH repeater layer,
+which manages the pipeline of case and form transactions received and manages the lifecycle of secure outbound
+messages to external systems.
+
+This architecture is designed to autonomously support the scale and volume of transactional data required for
+tens-to-hundreds of millions of transactions will be processed in a 24hr period.
```
[the remark this answers is not in the document]
Comment (snopoke) [reply docs/overview/platform.rst]: 465391348a0

**Review by snopoke — COMMENTED** [review]
@esoergel thanks for the detailed review. I confess I did not read all the text in detail when putting it in. I've made various updates.

File: docs/overview/architecture.rst [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+interminably long for an operation to happen. AMQP and the technologies surrounding it make for a clean, reusable
+interface to allow developers to create, execute, and retrieve results from these long running tasks.
+
+The python library that utilizes AMQP and RMQ is the Celery_ project, an open source library for asynchronous task
+queuing. A task can be written in python code to do a database operation or other report for CommCareHQ. To execute
+the task, the website can transmit a job request that is sent to the RabbitMQ queue. Separate worker processes on
+other dedicated machines can receive these tasks requests by querying the RabbitMQ server for new task requests.
+Once the worker completes the task, it can then notify the frontend of its completion in various ways. Either
+sending an email to the user making the request that the job is completed, and providing a link, or utilizing
+redis, updating the content of a URL the user is viewing to show that the task is completed.
```
[the remark this answers is not in the document]
Comment (esoergel) [reply docs/overview/architecture.rst]: :+1:

**Review by esoergel — APPROVED** [review]

**Pull request merged**